src/main.py

```diff
@@ -108,31 +51,14 @@ def main():
     print(table)
     print("\n")
 
-    while 1:
-        if keyboard.is_pressed(config.general['resume_key']):
-            screen = Screen(config.general["monitor"])
-            # Run health monitor thread
-            health_manager = HealthManager(screen)
-            health_monitor_thread = threading.Thread(target=health_manager.start_monitor)
-            health_monitor_thread.daemon = True
-            health_monitor_thread.start()
-            # Run death monitor thread
-            death_manager = DeathManager(screen)
-            death_monitor_thread = threading.Thread(target=death_manager.start_monitor)
-            death_monitor_thread.daemon = True
-            death_monitor_thread.start()
-            # Create other "global" instances
-            game_recovery = GameRecovery(screen, death_manager)
-            game_stats = GameStats()
-            run_bot(config, screen, game_recovery, game_stats, death_manager, health_manager)
-            break
-        if keyboard.is_pressed(config.general['auto_settings_key']):
-            adjust_settings()
-            break
-        elif keyboard.is_pressed(config.general['graphic_debugger_key']):
-            run_graphic_debugger()
-            break
-        time.sleep(0.02)
+    keyboard.add_hotkey(config.general['auto_settings_key'], lambda: adjust_settings(config))
+    keyboard.add_hotkey(config.general['graphic_debugger_key'], lambda: run_graphic_debugger())
+    keyboard.add_hotkey(config.general['restore_settings_from_backup_key'], lambda: restore_settings_from_backup(config))
+    keyboard.add_hotkey(config.general['settings_backup_key'], lambda: backup_settings(config))
+    keyboard.add_hotkey(config.general['resume_key'], lambda c: start_bot(c), args=[config])
+    keyboard.add_hotkey(config.general["exit_key"], lambda: Logger.info(f'Force Exit') or os._exit(1))
```

Comment:
why I did not do it this way is to avoid having graphic debugger run while botty is running and to not have multiple bottys run at the same time. The logic now is that the "resume_key" is overwritten within run_bot() and thus will not trigger another start_bot() as I see it. A bit hacky, but good enough for me.

Reply:
Fair enough, I now better understood your intention with the first code iteration and I agree that it felt hacky to re-bound the key, so I changed it to let the game controller take care of that. So if botty is running and you want to pause it, F11 will not pause the bot, if it is not running it will start it instead. That without removing the hotkey and re-adding it